lib/game.rb

```diff
@@ -0,0 +1,104 @@
+require_relative "players/human.rb"
+require_relative "players/computer.rb"
+require_relative "board.rb"
+
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ] 
+
+  def initialize(player_1=Human.new("X"), player_2=Computer.new("O"), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board  
+  end
+
+  def board
+    @board
+  end
+
+  def player_2
+    @player_2
+  end
+
+  def player_1
+    @player_1
+  end
+
+  def current_player
+    if ((@board.turn_count)+2) % 2 == 0
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      @board.position(combo[0]+1) == @board.position(combo[1]+1) &&
+      @board.position(combo[1]+1) == @board.position(combo[2]+1) &&
+      @board.taken?(combo[0]+1)
+    end
+  end
+
+  def draw?
+    @board.full? && won? == nil
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def winner
+    if won? != nil
+      @board.cells[won?.first]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.